### PR TITLE
Sacess fixes

### PR DESCRIPTION
**Pull request by dweindl**
* Fix TypeError for size-1-arrays
* Optional arguments
* logging
* optional multiprocessing instead of multithreading
* cleanup

### CODE REVIEW

File: pypesto/optimize/ess/ess.py [hunk 1]
```diff
@@ -109,6 +114,12 @@ def __init__(
         balance:
             Quality vs diversity balancing factor [0, 1];
             0 = only quality; 1 = only diversity
+        n_procs:
+            Number of parallel process to use for parallel function evaluation.
```
Comment (yannikschaelte) [inline pypesto/optimize/ess/ess.py:118]:
```suggestion
            Number of parallel processes to use for parallel function evaluation.
```

Comment (yannikschaelte) [reply pypesto/optimize/ess/ess.py:118]: Check mutual exclusivity somewhere?

**Review by yannikschaelte — APPROVED** [review]

File: pypesto/optimize/ess/ess.py (hunk 1 above)
Comment (dweindl) [reply pypesto/optimize/ess/ess.py:118]: Good point.

**Comment by PaulJonasJost** [comment]
@dweindl can this branch be deleted?

**Pull request merged**